### ISSUE TITLE
Receiving error "Cannot read properties of undefined" at method getClient

### ISSUE THREAD

**Issue by EduardoCandido** [Azure/azure-functions-durable-js#562]
**Describe the bug**
The error occurs at my start function and throws the error:

`2023-11-30T19:32:47.506 [Error] Executed 'Functions.HttpStart' (Failed, Id=bb0d5e80-7c64-487d-8b31-915efd8cbc76, Duration=5ms)Result: FailureException: Cannot read properties of undefined (reading 'extraInputs')Stack: TypeError: Cannot read properties of undefined (reading 'extraInputs')at Object.getClient (C:\home\site\wwwroot\node_modules\durable-functions\lib\src\durableClient\getClient.js:11:40)at module.exports (C:\home\site\wwwroot\HttpStart\index.js:4:23)at t.InvocationModel.<anonymous> (C:\Program Files (x86)\SiteExtensions\Functions\4.27.5\workers\node\dist\src\worker-bundle.js:2:59767)at Generator.next (<anonymous>)at C:\Program Files (x86)\SiteExtensions\Functions\4.27.5\workers\node\dist\src\worker-bundle.js:2:58092at new Promise (<anonymous>)at p (C:\Program Files (x86)\SiteExtensions\Functions\4.27.5\workers\node\dist\src\worker-bundle.js:2:57837)at t.InvocationModel.invokeFunction (C:\Program Files (x86)\SiteExtensions\Functions\4.27.5\workers\node\dist\src\worker-bundle.js:2:59574)at y.<anonymous> (C:\Program Files (x86)\SiteExtensions\Functions\4.27.5\workers\node\dist\src\worker-bundle.js:2:37872)at Generator.next (<anonymous>)`

**Investigative information**

- Durable Functions extension version: 4 
- durable-functions npm module version: 3
- Language (JavaScript/TypeScript) and version: Javascript
- Node.js version: 18

***If deployed to Azure App Service***

- Timeframe issue observed: 2023-11-30T19:40:58.558
- Function App name: durable-functionsaz204
- Function name(s): HttpStart
- Region: Brazil South
- Orchestration instance ID(s): bb0d5e80-7c64-487d-8b31-915efd8cbc76

**To Reproduce**

1. Go to HttpStart Function URL and send a request

This is the start function code
```
const df = require("durable-functions");

module.exports = async function (context, req) {
    const client = df.getClient(context);
    const instanceId = await client.startNew(req.params.functionName, undefined, req.body);

    context.log(`Started orchestration with ID = '${instanceId}'.`);

    return client.createCheckStatusResponse(context.bindingData.req, instanceId);
};
```

**Screenshots**
![image](https://github.com/Azure/azure-functions-durable-js/assets/38570289/27a2996c-4461-4f5f-867c-4e191cef414c)

**Known workarounds**
Provide a description of any known workarounds you used.

**Additional context**

Could be helpfull to reproduce, I was following this tutorial https://learn.microsoft.com/en-us/training/modules/create-long-running-serverless-workflow-with-durable-functions/4-exercise-create-a-workflow-using-durable-functions

**Comment by ejizba** [comment]
Hi @EduardoCandido I believe you have some mismatched versions here. I would recommend you downgrade your "durable-functions" package version to v2 (I assume you're currently using v3) or follow this guide to upgrade your code to match the latest package: https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-node-model-upgrade?tabs=nodejs-v4&pivots=programming-language-javascript

**Comment by EduardoCandido** [comment]
Dowgrade to v2 worked also I had to add the moment dependency on package.json, thanks a lot

**Comment by ISkomorokh** [comment]
@ejizba, thank you for helping resolve this.

Anyway, that is very weird since I'm following basic instructions from learn.microsoft.com, and it is a code from the Azure templates. https://learn.microsoft.com/en-us/training/modules/create-long-running-serverless-workflow-with-durable-functions/4-exercise-create-a-workflow-using-durable-functions

**Comment by chidijude** [comment]
> Dowgrade to v2 worked also I had to add the moment dependency on package.json, thanks a lot

How did you do this?

**Comment by EduardoCandido** [comment]
@chidijude 

1 - Access App Files on your Function App and in the file host.json change the version to v2
![image](https://github.com/Azure/azure-functions-durable-js/assets/38570289/d1a216b3-e38c-47f6-a3e5-9466c167d7df)

2 - Also in the package.json file be sure to use the durable-function in version 2 and add other dependencies you have to use
![image](https://github.com/Azure/azure-functions-durable-js/assets/38570289/b952c467-3172-415c-b72c-917cd8023079)

**Comment by prashantkasbe** [comment]
Once you have updated the package.json and host.json as mentioned by @EduardoCandido Go to Development Tools, select Console. The Console pane appears for your function app. and run npm install durable-functions.

You need to redeploy the packages to make the call work and the changes to take effect.

**Comment by indrajithekanayake** [comment]
Once you do all the above go to Overview in your Azure Function and hit Refresh in the top menu

**Comment by hmantri05** [comment]
@prashantkasbe I have a similar issue when I migrated my function app from Runtime Environment 3.x to 4.x

This [page](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-node-model-upgrade?tabs=nodejs-v4&pivots=programming-language-javascript) mentions that we have to use `durable-function 3.x` for v4 "programming model". So I used `durable-functions 3.0.0` in my `package.json`.

But the above comments suggest that we have to use durable-functions 2.x.

Does "Programming Model" have a different meaning than "Runtime Environment" ?
